tests/core/full_node/test_block_height_map.py

```diff
@@ -220,8 +243,21 @@ async def test_restore_extend(self, tmp_dir, db_version):
             await setup_db(db_wrapper)
             # add 2000 blocks to the chain
             await setup_chain(db_wrapper, 4000, ses_every=20)
+
+            # At this point we should have a proper cache with the old chain data
+            async with aiofiles.open(tmp_dir / "height-to-hash", "rb") as f:
+                heights = bytearray(await f.read())
+                assert len(heights) == (2000 + 1) * 32
+
             height_map = await BlockHeightMap.create(tmp_dir, db_wrapper)
 
+            # Make sure we properly wrote the additional data to the cache
+            async with aiofiles.open(tmp_dir / "height-to-hash", "rb") as f:
+                new_heights = bytearray(await f.read())
+                assert len(new_heights) == (4000 + 1) * 32
+                for i in range(0, len(heights), 32):
+                    assert new_heights[i : i + 32] == heights[i : i + 32]
```

Comment:
is `len(new_heights)` allowed to be different than `len(heights)`?
it looks like this might be able to be replaced by:
```
assert heights == await f.read()
```

Reply:
Here it's different because we're extending the initial 2000 DB items with 2000 more. The file cache gets updated accordingly.